### PR TITLE
ci: comment and close issues when stable, just comment on nightly

### PR DESCRIPTION
**Pull request by mcansh**
Signed-off-by: Logan McAnsh <logan@mcan.sh>

Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

https://github.com/mcansh/nightly-release-test

### CODE REVIEW

**Review by MichaelDeBoey — APPROVED** [review]

File: scripts/release/constants.ts [hunk 1]
```diff
@@ -30,3 +30,5 @@ export const GITHUB_REPOSITORY = process.env.GITHUB_REPOSITORY;
 export const DEFAULT_BRANCH = process.env.DEFAULT_BRANCH;
 export const NIGHTLY_BRANCH = process.env.NIGHTLY_BRANCH;
 export const PR_FILES_STARTS_WITH = ["packages/"];
+export const isNightlyRelease = isNightly(VERSION);
+export const AWAITING_RELEASE_LABEL = "awaiting release";
```
Comment (MichaelDeBoey) [inline scripts/release/constants.ts:34]: Should we add this label to each PR that's merged to `dev`?

Comment (mcansh) [reply scripts/release/constants.ts:34]: just thought about the flow some more and i think this is what i'm gonna go for. this way if a PR is reverted we don't prematurely close anything (unless of course a label is incorrectly removed or added)

1. PR is merged
  - add "awaiting release" label to PR and linked issues
2. nightly release comments on PR and linked issues
3. stable comes around
  a. find merged PRs
  b. comment on PRs that have "awaiting release" label
    - remove "awaiting release" label
  c. comment on linked issues
    - remove "awaiting release" label

Comment (MichaelDeBoey) [reply scripts/release/constants.ts:34]: Sounds good to me 👍

Comment (mcansh) [reply scripts/release/constants.ts:34]: actually, we don't need the label on issues, the PR label should be enough

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -49,14 +49,12 @@ jobs:
...
-          LATEST_NIGHTLY_TAG=$(git tag -l v0.0.0-nightly-\* --sort=-taggerdate | head -n 1)
+          LATEST_NIGHTLY_TAG=$(git tag -l v0.0.0-nightly-\* --sort=-committerdate | head -n 1)
 
-          # check if last commit to dev starts would be the nightly tag we're about to create (minus the date)
-          # if it is, we'll skip the nightly creation
-          # if not, we'll create a new nightly tag
-          if [[ ${LATEST_NIGHTLY_TAG} == v0.0.0-nightly-${SHORT_SHA}-* ]]; then
-            echo "🛑 Latest nightly tag is the same as the latest commit sha, skipping nightly release"
-          else
+          CHANGES=$(git diff $LATEST_NIGHTLY_TAG..dev -- ./packages/ -- ':!packages/**/package.json')
```
Comment (mcansh) [inline .github/workflows/nightly.yml:54]: check for changes to `packages/*` ignoring updates to `package.json` due to the version making the file change

**Pull request merged**